tests/src/Unit/PackageTest.php

```diff
@@ -17,7 +23,7 @@ public function testAvoidDuplicates() {
         $package1->addDependency($package2);
         $package2->addResolves($package1);
 
-        $this->assertEquals(1, count($package1->getDependencies()));
-        $this->assertEquals(1, count($package2->getResolves()));
+        static::assertSame(1, count($package1->getDependencies()));
+        static::assertSame(1, count($package2->getResolves()));
```

Comment:
What is your reasoning for `static::assert` over `$this->assert`?

Reply:
Yes, I know that `$this->assert*` is widely used. I have never understood why.
There is somewhere a post by Sebastian Bergmann both can be used. None of them is preferred over the other.
PHPStorm highlights `$this->assert*` as a warning, and I agree with that.
If the method defined as static why should I call it on an instance?
This is just my opinion, but just because the PHP language allows it not necessarily a good practice.

Reply:
Humm. That is another comment. It has a clear suggestion:
> A regular user of PHPUnit should use $this-> instead of self::

I am not fun of the `self::` either, but it is another story.
That comment was written in 2015. Maybe `static::` was not available back then. I don't remember.